src/m365/spo/commands/tenant/tenant-commandset-set.ts

```diff
@@ -0,0 +1,223 @@
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const appCatalogUrl = await spo.getTenantAppCatalogUrl(logger, this.debug);
+      if (!appCatalogUrl) {
+        throw 'No app catalog URL found';
+      }
+
+      await this.updateTenantWideExtension(appCatalogUrl, args.options, logger);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  public async getTenantCommandSet(logger: Logger, options: Options, requestUrl: string): Promise<number> {
+    if (this.verbose) {
+      logger.logToStderr(`Getting the tenant command set ${options.id}`);
+    }
+
+    const filter: string = `startswith(TenantWideExtensionLocation,'ClientSideExtension.ListViewCommandSet') and Id eq '${options.id}'`;
+
+    const reqOptions: CliRequestOptions = {
+      url: `${requestUrl}/items?$filter=${filter}`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const listItemInstances: ListItemInstanceCollection = await request.get<ListItemInstanceCollection>(reqOptions);
+
+    if (listItemInstances.value.length === 0) {
+      throw 'The specified command set was not found';
+    }
+
+    return listItemInstances.value[0].Id;
+  }
+
+  private async updateTenantWideExtension(appCatalogUrl: string, options: Options, logger: Logger): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr('Updating tenant wide extension to the TenantWideExtensions list');
+    }
+
+    const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
+    const requestUrl = `${appCatalogUrl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')`;
+    const id = await this.getTenantCommandSet(logger, options, requestUrl);
+
+    const commandOptions: spoListItemSetCommandOptions = {
+      id: id.toString(),
+      webUrl: appCatalogUrl,
+      listUrl: `${urlUtil.getServerRelativeSiteUrl(appCatalogUrl)}/Lists/TenantWideExtensions`
+    };
+
+    if (options.newTitle) {
+      commandOptions.Title = options.newTitle;
+    }
+
+    if (options.clientSideComponentId) {
+      commandOptions.TenantWideExtensionComponentId = options.clientSideComponentId;
+    }
+
+    if (options.location) {
+      commandOptions.TenantWideExtensionLocation = this.getLocation(options.location);
+    }
+
+    if (options.listType) {
+      commandOptions.TenantWideExtensionListTemplate = this.getListTemplate(options.listType);
+    }
+
+    if (options.clientSideComponentProperties) {
+      commandOptions.TenantWideExtensionComponentProperties = options.clientSideComponentProperties;
+    }
+
+    if (options.webTemplate) {
+      commandOptions.TenantWideExtensionWebTemplate = options.webTemplate;
+    }
+
+    await Cli.executeCommand(spoListItemSetCommand as Command, { options: { ...commandOptions, _: [] } });
```

Comment:
Let's not use this as we are in the process of moving away from using other commands